gui/webdisplay/src/TWebWindow.cxx

```diff
@@ -407,13 +447,35 @@ void ROOT::Experimental::TWebWindow::Send(const std::string &data, unsigned conn
          SendDataViaConnection(*iter, chid, data);
       } else {
          assert(iter->fQueue.size() < fMaxQueueLength && "Maximum queue length achieved");
-         iter->fQueue.push_back(std::to_string(chid) + ":" + data);
+         iter->fQueue.emplace_back(chid, data);
+      }
+   }
+
+   CheckDataToSend();
+}
+
+///////////////////////////////////////////////////////////////////////////////////
+/// Send binary data to specified connection
+/// If connid==0, data will be send to all connections
+
+void ROOT::Experimental::TWebWindow::SendBinary(std::shared_ptr<RawBuffer> &data, unsigned connid, unsigned chid)
+{
+   for (auto iter = fConn.begin(); iter != fConn.end(); ++iter) {
+      if (connid && connid != iter->fConnId)
+         continue;
+
+      if ((iter->fQueue.size() == 0) && (iter->fSendCredits > 0)) {
+         SendBinaryDataViaConnection(*iter, chid, data);
+      } else {
+         assert(iter->fQueue.size() < fMaxQueueLength && "Maximum queue length achieved");
```

Comment:
Here and in other places: `assert()` means that the program will stop if this is hit. Usually that's not worth it. I would very much prefer `R__FATAL_HERE` or even only a `R__ERROR_HERE`, possibly protected with `#ifndef NDEBUG` for "CPU intense" checks.